tests/Microsoft.Identity.Web.Test/TokenAcquisitionAuthorityTests.cs

```diff
@@ -309,5 +310,86 @@ public void MergeExtraQueryParameters_MergedOptionsNull_Test()
             // Assert
             Assert.Null(mergedDict);
         }
+
+        [Theory]
+        [MemberData(nameof(ManagedIdClientIdCases))]
+        public void ManagedIdCacheKey_Test(string clientId)
+        {
+            // Arrange
+            string defaultKey = ;
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+
+            // Act
+            string key = GetCacheKeyForManagedIdReflection(managedIdentityOptions);
+
+            // Assert
+            if (string.IsNullOrEmpty(clientId))
+            {
+                Assert.Equal(defaultKey, key);
+            }
+            else
+            {
+                Assert.Equal(clientId, key);
+            }
+        }
+
+
+        [Theory]
+        [MemberData(nameof(ManagedIdClientIdCases))]
+        public void GetOrBuildManagedIdentity_Test(string clientId)
+        {
+            // Arrange
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+            BuildTheRequiredServices();
+            MergedOptions mergedOptions = new();
+            InitializeTokenAcquisitionObjects();
+            string key = GetCacheKeyForManagedIdReflection(managedIdentityOptions);
+            ConcurrentDictionary<string, IManagedIdentityApplication?> cacheDict = GetManagedIdCacheReflection();
+
+            // Act
+            IManagedIdentityApplication app = 
```

Comment:
id avoid using .Result even in tests, just make the Test method `async Task`

Reply:
Removed it, out of curiosity, what is wrong with using .Result?

Reply:
It is async over sync.